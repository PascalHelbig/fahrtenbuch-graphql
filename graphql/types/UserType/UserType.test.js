const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const UserType = require('./UserType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: User }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, UserType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      User: () => ({
        id: () => '1337',
        name: () => 'user name',
      }),
    },
  });

  const query = `
    query {
      id
      name
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot(),
  );
});
