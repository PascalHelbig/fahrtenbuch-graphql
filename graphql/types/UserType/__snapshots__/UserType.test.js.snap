exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "id": "1337",
    "name": "user name",
  },
}
`;
